org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/workspace/ContextActivatingSelectionListener.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2015 the original author or authors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Etienne Studer & Donát Csikós (Gradle Inc.) - initial API and implementation and initial documentation
+ */
+
+package org.eclipse.buildship.ui.workspace;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Predicate;
+
+import org.eclipse.jface.viewers.ISelection;
+import org.eclipse.ui.ISelectionListener;
+import org.eclipse.ui.IViewPart;
+import org.eclipse.ui.IWorkbenchPart;
+import org.eclipse.ui.IWorkbenchPartReference;
+import org.eclipse.ui.contexts.IContextActivation;
+import org.eclipse.ui.contexts.IContextService;
+import org.eclipse.ui.services.IServiceLocator;
+
+/**
+ * Dynamically activates the context with the specified id whenever the currently active
+ * {@link IWorkbenchPartReference} belongs to the specified {@link IViewPart}.
+ */
+public final class ContextActivatingSelectionListener implements ISelectionListener {
```

Comment:
We should at least store the ContextActivatingSelectionListener and the ContextActivatingViewPartListener class at the same package.